SHTSensor.cpp

```diff
@@ -203,6 +203,55 @@ class SHT3xSensor : public SHTI2cSensor
 };
 
 
+//
+// class SHT4xSensor
+//
+
+class SHT4xSensor : public SHTI2cSensor
+{
+private:
+  static const uint16_t SHT4X_ACCURACY_HIGH    = 0xFD00;
+  static const uint16_t SHT4X_ACCURACY_MEDIUM  = 0xF600;
+  static const uint16_t SHT4X_ACCURACY_LOW     = 0xE000;
+
+  static const uint8_t SHT4X_ACCURACY_HIGH_DURATION   = 10;
+  static const uint8_t SHT4X_ACCURACY_MEDIUM_DURATION = 4;
+  static const uint8_t SHT4X_ACCURACY_LOW_DURATION    = 2;
+
+public:
+  static const uint8_t SHT4X_I2C_ADDRESS_44 = 0x44;
+  static const uint8_t SHT4X_I2C_ADDRESS_45 = 0x45;
+
+  SHT4xSensor(uint8_t i2cAddress = SHT4X_I2C_ADDRESS_44)
+      : SHTI2cSensor(i2cAddress, SHT4X_ACCURACY_HIGH,
+                     SHT4X_ACCURACY_HIGH_DURATION,
+                     -45, 175, 65535, 125, 65535,-6,1)
```

Comment:
why is cmdSize 1 here? 0xFD00, 0xF600, 0xE000 et al are all two bytes?

Reply:
The reason for the cmd size varible was that the SHT4x commands are in fact only one byte. The function readSample() from the SHTI2CSensor base class however used the hard coded CMD Size of two byte. So in order to make the function more flexible I introduced the variable. I defined the commands as two byte commands (i.e. 0xFD00) because they are also shifted in the readSample function. This way the older sensors and the SHT4x can use the same functions and that was in my opinion the easiest implementation where not too much has to be changed. If you have any other ideas please let me know

Reply:
ah, makes sense. I think we'll probably want to readSample() down the line, but I think that can be a refactoring later